kuzzleio/kuzzle
# Allow to dynamically add strategy in the plugin init method

## What does this PR do ?

When dynamically adding strategies with the `accessors.strategies.add` method in the plugin `init` method, the authenticators have not been initialized yet.  

This PR initialize statically declared resources (controllers, authenticators, pipes, etc.) before calling the plugin `init` method.

https://deploy-preview-1479--doc-kuzzle.netlify.com/plugins/guides/strategies/overview/

### How should this be manually tested?

Test plugin:
```
class TestPlugin {
  constructor () {
    this.authenticators = {
      'ExampleStrategy': require('passport-custom').Strategy
    };
  }

  init (config, context) {
    return context.accessors.strategies.add('test',  {
      config: {
        authenticator: 'ExampleStrategy'
      },
      methods: {
        create: 'create',
        delete: 'delete',
        exists: 'exists',
        getById: 'getById',
        getInfo: 'getInfo',
        update: 'update',
        validate: 'validate',
        verify: 'verify'
      }
    })
  }

  create () {}
  exists () {}
  delete () {}
  getById () {}
  getInfo () {}
  update () {}
  validate () {}
  verify () {}
}

module.exports = TestPlugin
```
